src/main/java/com/redhat/devtools/intellij/tektoncd/tkn/TknCli.java

```diff
@@ -1117,6 +1126,81 @@ private String getFormattedWarningMessage(String reason, String message) {
         return data + "\n";
     }
 
+    public List<String> getResourcesAsYaml(List<Resource> resources) throws IOException {
```

Comment:
Not sure to understand the rationale for using CompletableFuture as you're synchronizing later to get the result.

Reply:
To make calls to cluster async as you may try to download 10 different resources at the same time. Then i need to sync them up as i need all of them to create the bundle... but atleast the download part should be faster than dong it sync, no?